content/en/docs/architecture/timeouts.md

```diff
@@ -1,24 +1,55 @@
+
+Timeouts occur for a job when the overall job or some subset of the job exceeds the configured budget for execution. Preemptions occur when a new version of the job is triggered, and the previous version is rendered obsolete, as happens when new code is pushed to a Pull Request and previous commits are no longer important to test. Timeouts and preemptions are implemented via `Pod` deletions on our build farm clusters. Therefore, all interruption events will obey the basic rules of Kubernetes [graceful termination](https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-terminating-with-grace): every process is given a `SIGTERM`, some time to react to it gracefully, then given a `SIGKILL`.
+
+Furthermore, instability in the OpenShift build farm clusters hosting these workloads may also trigger the deletion of `Pods` that implement a job. In these cases, the resulting workflow is identical, as `Pod` eviction honors the configured grace periods.
+
+### Types of Timeouts
+
+#### Prow Infrastructure Timeouts
+
+Prow is furthermore [configured](https://github.com/openshift/release/blob/9238ee8b89c861793f9b8e8f7f6509abe33bd0b8/core-services/prow/02_config/_config.yaml) to tightly bound the execution time budgets for all jobs in the system. The configuration specifies a number of timeouts, exceeding any of these will cause the test `Pod` (running `ci-operator`) to be deleted, triggering the Kubernetes graceful termination process. Each timeout has specific semantics:
+
+- The `default_job_timeout` is a hard timeout for the overall lifetime of a job. This may be overridden for a specific job, but only to values smaller than the global timeout. This setting is global for all Prow jobs, it's needed as Prow can launch job types other than Kubernetes `Pods`.
+- The `plank.pod_{pending,unscheduled}_timeout` settings allow Prow to move on from a job for which `Pods` cannot start. Jobs exhibiting these timeouts occur when issues exist with the underlying build farm clusters where `Pods` are executed. These jobs will be listed as errored, not failed, and the DPTP on-call rotation will investigate them.
+- The `plank.pod_running_timeout` is the default timeout for the overall lifetime for a running `Pod` for a job. This value is _not_ encoded in the `pod.spec.activeDeadlineSeconds`, but handled by Prow itself.
```

Comment:
This would only do anything if the decoration utils timeout doens't work which would basically be a bug or not?